lib/utils.js

```diff
@@ -538,6 +538,23 @@ exports.populate = function populate(path, select, model, match, options, subPop
     throw new TypeError('utils.populate: invalid path. Expected string. Got typeof `' + typeof path + '`');
   }
 
+  if (Array.isArray(subPopulate)) {
+    let ret = [];
+    subPopulate.forEach(function(obj) {
+      if (/[\s]/.test(obj.path)) {
+        let copy = exports.object.shallowCopy(obj);
```

Comment:
I'd just use `Object.assign({}, obj)`, more standard way to do shallow cloning.

Reply:
fixed.